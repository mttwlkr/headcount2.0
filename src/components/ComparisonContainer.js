import React from 'react';
import ComparisonCard from './ComparisonCard'
import Card from './Card'
import PropTypes from 'prop-types';

const ComparisonContainer = ({ cards, handleSelection }) => {

  const [cardA, comparisonCard, cardB] = cards
  
  return (
    <section>
      <Card info={cardA} handleSelection={handleSelection} />
      { comparisonCard && <ComparisonCard comparisonCard={comparisonCard} />}
      { comparisonCard && <Card info={cardB} handleSelection={handleSelection} />}
    </section>
  )
}

ComparisonContainer.propTypes = {
  cards: PropTypes.array
}

export default ComparisonContainer;
