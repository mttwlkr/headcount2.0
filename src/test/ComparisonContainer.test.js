import React from 'react'
import { shallow, mount } from 'enzyme'
import ComparisonContainer from '../components/ComparisonContainer'

describe('Comparison Container', () => {

  it('should match the snapshot', () => {
    const comparisonCard = {COLORADO: 0.53, ACADEMY: 0.407, compared: 1.302}
    const clearedComparison = jest.fn() 
    const generateComparisons = jest.fn()

    const wrapper = shallow(<ComparisonContainer 
      cards={}  
      clearedComparison={clearedComparison} 
      />)

    cards, handleComparison, clearedComparison, generateComparisons

    expect(wrapper).toMatchSnapshot()
  })

})