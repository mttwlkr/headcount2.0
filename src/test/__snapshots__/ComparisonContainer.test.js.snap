// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    cards={
        Array [
            Object {
              "data": Object {
                "2006": 0.098,
              },
              "location": "Denver",
            },
            Object {
              "data": Object {
                "2006": 0.848,
              },
              "location": "Golden",
            },
          ]
    }
    clearedComparison={[Function]}
    generateComparisons={[Function]}
    handleComparison={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          handleComparison={[Function]}
          info={
                    Object {
                              "data": Object {
                                "2006": 0.098,
                              },
                              "location": "Denver",
                            }
          }
/>,
        <ComparisonCard
          clearedComparison={[Function]}
          comparisonCard={undefined}
/>,
        <Card
          handleComparison={[Function]}
          info={
                    Object {
                              "data": Object {
                                "2006": 0.848,
                              },
                              "location": "Golden",
                            }
          }
/>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleComparison": [Function],
          "info": Object {
            "data": Object {
              "2006": 0.098,
            },
            "location": "Denver",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clearedComparison": [Function],
          "comparisonCard": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleComparison": [Function],
          "info": Object {
            "data": Object {
              "2006": 0.848,
            },
            "location": "Golden",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            handleComparison={[Function]}
            info={
                        Object {
                                    "data": Object {
                                      "2006": 0.098,
                                    },
                                    "location": "Denver",
                                  }
            }
/>,
          <ComparisonCard
            clearedComparison={[Function]}
            comparisonCard={undefined}
/>,
          <Card
            handleComparison={[Function]}
            info={
                        Object {
                                    "data": Object {
                                      "2006": 0.848,
                                    },
                                    "location": "Golden",
                                  }
            }
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleComparison": [Function],
            "info": Object {
              "data": Object {
                "2006": 0.098,
              },
              "location": "Denver",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clearedComparison": [Function],
            "comparisonCard": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleComparison": [Function],
            "info": Object {
              "data": Object {
                "2006": 0.848,
              },
              "location": "Golden",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
